terminal/runereader.go

```diff
@@ -34,43 +34,48 @@ func (rr *RuneReader) ReadLine(mask rune) ([]rune, error) {
 	// we only care about horizontal displacements from the origin so start counting at 0
 	index := 0
 
+	cursor := &Cursor{
+		In:  rr.stdio.In,
+		Out: rr.stdio.Out,
+	}
+
 	// we get the terminal width and height (if resized after this point the property might become invalid)
-	terminalSize, _ := Size()
+	terminalSize, _ := cursor.Size(rr.state.buf)
 	for {
 		// wait for some input
 		r, _, err := rr.ReadRune()
 		if err != nil {
 			return line, err
 		}
 		// we set the current location of the cursor and update it after every key press
-		cursorCurrent, err := CursorLocation()
+		cursorCurrent, err := cursor.CursorLocation(rr.state.buf)
```

Comment:
Small nitpick. Since the cursor functions are now methods of a struct, can we remove the `Cursor` at the front of all of these? So this would look like `cursor.Location(rr.state.buf)`?

Reply:
I agree. I've also been thinking about an alternative where `CursorUp` and the like simply return `string` so that the caller can write to appropriate writer, but `CursorLocation` will still need to interact with the tty directly. If we stay with the `struct` approach I definitely think we should remove the redundant `Cursor` in the methods.